orchagent/intfsorch.h

```diff
@@ -12,20 +12,66 @@
 
 #include <map>
 #include <set>
+#include <unordered_map>
+
 
 extern sai_object_id_t gVirtualRouterId;
 extern MacAddress gMacAddress;
 
 #define RIF_STAT_COUNTER_FLEX_COUNTER_GROUP "RIF_STAT_COUNTER"
 
+
 struct IntfsEntry
 {
     std::set<IpPrefix>  ip_addresses;
     int                 ref_count;
+
+    IntfsEntry(int rc = 0) : ref_count(rc) {};
+    IntfsEntry(IpPrefix p)
+    {
+        ip_addresses.insert(p);
+        ref_count++;
+    }
 };
 
 typedef map<string, IntfsEntry> IntfsTable;
 
+struct IntfRouteEntry
+{
+    IpPrefix prefix;
+    string   ifName;
+    string   type;
+
+    IntfRouteEntry(const IpPrefix &p, string n, string t = "subnet")
+        : prefix(p), ifName(n), type(t) {};
+
+    inline bool operator==(const IntfRouteEntry &x) const
+    {
+        return (prefix == x.prefix && ifName == x.ifName);
+    }
+};
+
+/*
+ * Hashmap to keep track of all interface-specific routes in the system.
+ * Indexed by the string associated to each interface-route (either ip2me or
+ * subnet or bcast). Values are formed by a list of elements that keep track
+ * of each route IpPrefix, as well as the interface on which it was configured.
+ *
+ * Example:
+ *
+ *    Key                                        Value
+ * ----------             -------------------------------------------------------
+ * 10.1.1.0/24 (subnet)   10.1.1.0/24 eth1, 10.1.1.0/24 eth2, 10.1.1.0/24 eth3
+ * 10.1.1.1/32 (ip2me)    10.1.1.1/32 eth1, 10.1.1.10/32 eth2, 10.1.1.255/32 eth3
```

Comment:
>10.1.1.10/32 eth2, 10.1.1.255/32 eth3 [](start = 44, length = 37)

10.1.1.10 or 10.1.1.1?

Reply:
@lguohan you right, there's a typo here, it should be '10.1.1.1/32' in that second line, will adjust this.